dev/bots/check_tests_cross_imports.dart

```diff
@@ -0,0 +1,563 @@
+    }).toList();
+  }
+
+  /// Returns the Set of Files that are not in knownPaths.
+  static Set<File> _getUnknowns(Set<String> knownPaths, Set<File> files) {
+    return files.where((File file) {
+      final prefix = RegExp('packages.flutter.test');
+      final int index = file.absolute.path.indexOf(prefix);
+      if (index < 0) {
+        throw ArgumentError('All files must include $prefix in their path.', 'files');
+      }
+      final String comparablePath = file.absolute.path
+          .substring(index)
+          .replaceAll('/', Platform.isWindows ? r'\' : '/');
+      return !knownPaths.contains(comparablePath);
+    }).toSet();
+  }
+
+  // Get a list of all the filenames in the source directory that end in "_test.dart".
+  static Set<File> _getTestFiles(Directory directory, _Library library) {
+    return _getFiles(directory.childDirectory(library.directory), RegExp(r'_test\.dart$')).toSet();
+  }
+
+  /// Returns true only if the file imports the given Library.
+  static bool _containsImport(File testFile, _Library library) {
+    final String contents = testFile.readAsStringSync();
+    return contents.contains(library.import);
+  }
+
+  /// Returns a Set of all Files that import the given Library.
+  static Set<File> _getFilesWithImports(Set<File> testFiles, _Library library) {
+    final filesWithCrossImports = <File>{};
+    for (final testFile in testFiles) {
+      if (_containsImport(testFile, library)) {
+        filesWithCrossImports.add(testFile);
+      }
+    }
+    return filesWithCrossImports;
+  }
+
+  /// Returns the error message for the given known paths that no longer have a
+  /// cross import.
+  ///
+  /// `library` must not be `Library.Material`, because Material is allowed to
+  /// cross-import.
+  static String _getFixedImportError(Set<String> fixedPaths, _Library library) {
+    assert(fixedPaths.isNotEmpty);
+    final buffer = StringBuffer(
+      'Huzzah! The following tests in ${library.name} no longer contain cross imports!\n',
+    );
+    for (final path in fixedPaths) {
+      buffer.writeln('  $path');
+    }
+    final String knownListName = switch (library) {
+      _Library.widgets => 'knownWidgetsCrossImports',
+      _Library.cupertino => 'knownCupertinoCrossImports',
+      _Library.material => throw UnimplementedError(
+        'Material is responsible for testing its interactions with Cupertino, so it is allowed to cross-import.',
+      ),
+    };
+    buffer.writeln('However, they now need to be removed from the');
+    buffer.write('$knownListName list in the script /dev/bots/check_tests_cross_imports.dart.');
+    return buffer.toString().trimRight();
+  }
+
+  /// Returns the File's relative path.
+  String _getRelativePath(File file, [Directory? root]) {
+    root ??= flutterRoot;
+    return path.relative(file.absolute.path, from: root.absolute.path);
+  }
+
+  /// Returns the import error for the `files` in testLibraryName which import
+  /// importedLibraryName.
```

Comment:
should this also have a doc comment about the `testLibraryName` not being `Material` because it allows cross imports.

Reply:
I refactored it to only accept valid import errors, and I added a doc comment about what that means.